### PR TITLE
add cron tests for sample app

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1234

### Change description ###

add cron tests for sample app

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: k8s/aat/common/bsp/bulk-scan-sample-app.yaml [hunk 1]
```diff
@@ -21,6 +21,32 @@ spec:
...
+        serviceAccountName: tests-service-account
+        keyVaults:
+          bulk-scan:
+            resourceGroup: bulk-scan
+            secrets:
+              sample-app-s2s-secret: S2S_SECRET
+        environment:
+          TEST_URL: "http://bulk-scan-sample-app-aat.service.core-compute-aat.internal"
+          SLACK_CHANNEL: "bsp-test-notices"
+          SLACK_NOTIFY_SUCCESS: false
```
Comment (lgonczar) [inline k8s/aat/common/bsp/bulk-scan-sample-app.yaml:35]: Best to set it to `true`, see success notifications and then set to `false`

File: k8s/aat/common/bsp/bulk-scan-sample-app.yaml [hunk 2]
```diff
@@ -21,6 +21,32 @@ spec:
...
+          TEST_S2S_NAME: "bulk_scan_sample_app_tests"
+          TEST_S2S_URL: "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
+          TEST_SCAN_DELAY: "4000"
+          RUN_END_TO_END_TESTS: true
```
Comment (lgonczar) [inline k8s/aat/common/bsp/bulk-scan-sample-app.yaml:39]: This variable doesn't apply here. Actually, after the decision to have a separate project for end-to-end tests, it won't apply anywhere.

File: k8s/aat/common/bsp/bulk-scan-sample-app.yaml [hunk 3]
```diff
@@ -21,6 +21,32 @@ spec:
...
+        functionaltestscron:
+          enabled: true
+          image: hmctspublic.azurecr.io/bulk-scan/sample-app:prod-49b7c377
+          schedule: "15/30 6-19 * * mon-fri"
```
Comment (lgonczar) [inline k8s/aat/common/bsp/bulk-scan-sample-app.yaml:43]: We run functional test on schedule only for projects with end-to-end tests. And now that we're not going to have them in any of the services, we'll remove this bit from each config.

File: k8s/aat/common/bsp/bulk-scan-sample-app.yaml [hunk 4]
```diff
@@ -21,6 +21,32 @@ spec:
...
+        smoketests:
+          enabled: true
+          image: hmctspublic.azurecr.io/bulk-scan/sample-app:prod-49b7c377
```
Comment (lgonczar) [inline k8s/aat/common/bsp/bulk-scan-sample-app.yaml:46]: `.../sample-app-tests:...`? Here and below.

File: k8s/aat/common/bsp/bulk-scan-sample-app.yaml [hunk 5]
```diff
@@ -21,6 +21,31 @@ spec:
...
+          TEST_URL: "http://bulk-scan-sample-app-aat.service.core-compute-aat.internal"
+          SLACK_CHANNEL: "bsp-test-notices"
+          SLACK_NOTIFY_SUCCESS: true
+          TEST_S2S_NAME: "bulk_scan_sample_app_tests"
+          TEST_S2S_URL: "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
+          TEST_SCAN_DELAY: "4000"
+        functionaltestscron:
+          enabled: true
+          image: hmctspublic.azurecr.io/bulk-scan/sample-app-test:prod-49b7c377
+          schedule: "15/30 6-19 * * mon-fri"
```
Comment (lgonczar) [inline k8s/aat/common/bsp/bulk-scan-sample-app.yaml:42]: This bit should go

Comment (mdayican) [reply k8s/aat/common/bsp/bulk-scan-sample-app.yaml]: I misunderstood.

File: k8s/aat/common/bsp/bulk-scan-sample-app.yaml [hunk 6]
```diff
@@ -21,6 +21,30 @@ spec:
       replicas: 2
       useInterpodAntiAffinity: true
       image: hmctspublic.azurecr.io/bulk-scan/sample-app:prod-3b895c9b
+      testsConfig:
+        memoryLimits: "3072Mi"
```
Comment (nitinprabhu) [inline k8s/aat/common/bsp/bulk-scan-sample-app.yaml:25]: may be this much memory is not needed ?

File: k8s/aat/common/bsp/bulk-scan-sample-app.yaml (hunk 5 above)
Comment (doncem) [reply k8s/aat/common/bsp/bulk-scan-sample-app.yaml]: this block should go, not cron definition ;)

File: k8s/aat/common/bsp/bulk-scan-sample-app.yaml (hunk 6 above)
Comment (mdayican) [reply k8s/aat/common/bsp/bulk-scan-sample-app.yaml]: removed,
we do not set devmode so will use default .

**Pull request merged**